src/mbgl/renderer/style_diff.cpp

```diff
@@ -43,14 +38,27 @@ StyleDifference<T> diff(const std::vector<T>& a, const std::vector<T>& b) {
     return result;
 }
 
+ImageDifference diffImages(const std::vector<ImmutableImage>& a,
+                           const std::vector<ImmutableImage>& b) {
+    return diff(a, b, [] (const ImmutableImage& lhs, const ImmutableImage& rhs) {
+        return lhs->id == rhs->id;
```

Comment:
Cleaned this up in 4a963d40c934948a52406642f97dc77095a74a2f but left the assertion; I don't believe there's currently a way to trigger it.